src/go/configinfo/service_info.go

```diff
@@ -47,8 +47,12 @@ type ServiceInfo struct {
 	Operations []string
 	// Stores all methods info for this service, using selector as key.
 	Methods map[string]*methodInfo
-	// Stores url segment names, mapping snake name to Json name.
-	SegmentNames []*pmpb.SegmentName
+
+	// Stores the request type url for each selector.
+	RequestTypeNames map[string]string
+	// Stores mapping of snake_case to jsonCase segments for all fields per selector.
+	SegmentNames map[string]SnakeToJsonSegments
```

Comment:
should we move these two into MethodInfo?  They are for each selector.

Reply:
+1

Reply:
Good idea, done